lib/use-checkout-pricing.js

```diff
@@ -38,20 +47,39 @@ export default function useCheckoutPricing(initialInputs = {}, handleError = thr
   const recurly = useRecurly();
   const [loading, setLoading] = useState(false);
   const [input, setInput] = useState(initialInputs);
-  let checkoutPricing = recurly.Pricing.Checkout();
-  const [pricing, setPricing] = useState(checkoutPricing);
+  const [pricing, setPricing] = useState(recurly.Pricing.Checkout());
 
   useEffect(() => {
     setLoading(true);
-    const { subscriptions = [] } = input;
+    let checkoutPricing = recurly.Pricing.Checkout();
+    const { subscriptions = [], ...singleInputs } = input;
+
+    checkoutPricing = addSingleInputs(singleInputs, checkoutPricing);
+
     addSubscriptions(subscriptions, checkoutPricing)
       .then(() => {
-        setPricing(checkoutPricing);
+        checkoutPricing = checkoutPricing.reprice().done(() => {
+          setPricing(checkoutPricing);
+          setLoading(false);
+        });
       })
-      .catch(handleError)
-      .finally(() => setLoading(false));
+      .catch(error => {
+        handleError(error);
+        setLoading(false);
```

Comment:
for flow simplicity, `setLoading` can move into a final `then` or `done` below these two methods.

```js
addSubscriptions(subscriptions, checkoutPricing)
  .then(() => checkoutPricing = checkoutPricing.done(() => setPricing(checkoutPricing)))
  .catch(handleError)
  .then(() => setLoading(false));
```

Reply:
I tried making this change, but in this case, setLoading(false) fires before `setPricing` is called, causing the demo to flash `$0.00` before updating to its resolved value `$10.00`

Reply:
Will it respect the promise order if you use `checkoutPricing.then` instead of `checkoutPricing.done` ?

Reply:
I just tried this, but got the same result -- sets `loading` to `true` before the promise resolves. We can pair on it tomorrow.